AzureIoTHub.agent.lib.nut

```diff
@@ -629,513 +632,819 @@ class AzureIoTHub {
-                "sharedAccessExpiry": sas.se,
-                "connectionString": deviceConnectionString
-            };
+            foreach (reqId, cb in _msgSentCbMap) {
+                // TODO: Should we use imp.wakeup here?
+                _callOnComplete(cb, AZURE_IOT_CLIENT_ERROR_NOT_CONNECTED);
+            }
+            // TODO: Is it ok? Or we should delete every slot of this map
+            _msgSentCbMap = {};
 
-            return config;
+            if (_msgEnabledCb != null) {
+                // TODO: Should we use imp.wakeup here?
+                _callOnComplete(_msgEnabledCb, AZURE_IOT_CLIENT_ERROR_NOT_CONNECTED);
+                _msgEnabledCb = null;
+            }
+            if (_twinEnabledCb != null) {
+                // TODO: Should we use imp.wakeup here?
+                _callOnComplete(_twinEnabledCb, AZURE_IOT_CLIENT_ERROR_NOT_CONNECTED);
+                _twinEnabledCb = null;
+            }
+            if (_twinRetrCb != null) {
+                local cb = _twinRetrCb[1];
+                local timer = _twinRetrCb[2];
+                _twinRetrCb = null;
+                if (timer != null) {
+                    imp.cancelwakeup(timer);
+                }
+                try {
+                    // TODO: Should we use imp.wakeup here?
+                    cb(AZURE_IOT_CLIENT_ERROR_NOT_CONNECTED, null, null);
+                } catch (e) {
+                    _error("Exception at onRetrieve user callback: " + e);
+                }
+            }
+            foreach (reqId, arr in _twinUpdCbMap) {
+                local cb = arr[0];
+                local timer = arr[1];
+                if (timer != null) {
+                    imp.cancelwakeup(timer);
+                }
+                // TODO: Should we use imp.wakeup here?
+                _callOnComplete(cb, AZURE_IOT_CLIENT_ERROR_NOT_CONNECTED);
+            }
+            // TODO: Is it ok? Or we should delete every slot of this map
+            _twinUpdCbMap = {};
+            if (_dMethodEnabledCb != null) {
+                // TODO: Should we use imp.wakeup here?
+                _callOnComplete(_dMethodEnabledCb, AZURE_IOT_CLIENT_ERROR_NOT_CONNECTED);
+                _dMethodEnabledCb = null;
+            }
         }
 
-        function connect(done = null, onDisconnect = null) {
-            _transport.connect(done, onDisconnect);
-            return this;
+        // Check HTTP status
+        function _statusIsOk(status) {
+            return status / 100 == 2;
         }
 
-        function disconnect() {
-            _transport.disconnect();
-            return this;
+        // Metafunction to return class name when typeof <instance> is run
+        function _typeof() {
+            return "AzureIoTHubMQTTClient";
         }
 
-        function sendEvent(message, done = null) {
-            _transport.sendEvent(message, done);
-            return this;
+        function _logMsg(message, topic) {
```

Comment:
I would rather make one call to `_log` instead of four.

Reply:
(by building one single string)

Reply:
Done